ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -1162,14 +1167,12 @@ FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws St
                   + entries.size());
           Offset offsetEnd = offsetToStart;
           long currentTotalSizeOfEntries = 0;
+          Offset endOffsetOfSnapshot = getCurrentEndOffset(indexSegments);
           for (JournalEntry entry : entries) {
```

Comment:
nit: Now we are no this, I think we can do a de-dup before going through the for loop.

Reply:
addressed this with caching of messageinfos